tfjs-backend-webgpu/src/to_pixels_webgpu.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {DataType} from '@tensorflow/tfjs-core';
+
+import {getMainHeaderString as main, PixelsOpType, WebGPUProgram} from './webgpu_program';
+import {computeDispatch, flatDispatchLayout} from './webgpu_util';
+
+export class ToPixelsProgram implements WebGPUProgram {
+  variableNames = ['Image'];
+  uniforms = 'alpha: f32,';
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  workgroupSize: [number, number, number] = [64, 1, 1];
+  type: DataType;
+  textureFormat: GPUTextureFormat;
+  pixelsOpType = PixelsOpType.TO_PIXELS;
+  size = true;
+  alphaMode = false;
+
+  constructor(
+      outShape: number[], type: DataType, textureFormat: GPUTextureFormat,
+      alphaMode: boolean) {
+    this.outputShape = outShape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workgroupSize);
+    this.type = type;
+    this.textureFormat = textureFormat;
+    this.alphaMode = alphaMode;
+    this.shaderKey = `toPixels_${type}_${alphaMode}`;
+  }
+
+  getUserCode(): string {
+    let calculateResult;
+    if (this.type === 'float32') {
+      calculateResult = `
+      if (uniforms.numChannels == 1) {
+        rgba[0] = value;
+        rgba[1] = value;
+        rgba[2] = value;
+      } else {
+        rgba[d] = value;
+      }`;
+    } else {
+      calculateResult = `
+      if (uniforms.numChannels == 1) {
+        rgba[0] = value / 255.0;
+        rgba[1] = value / 255.0;
+        rgba[2] = value / 255.0;
+      } else {
+        rgba[d] = value / 255.0;
+      }`;
+    }
+    const alphaStr = this.alphaMode ? 'rgba[3] = uniforms.alpha;' : '';
+
+    const userCode = `
+       @group(0) @binding(0) var outImage : texture_storage_2d<${
+        this.textureFormat}, write>;
+       ${main('index')} {
+         if (index < uniforms.size) {
+           var rgba = vec4<f32>(0.0, 0.0, 0.0, 1.0);
```

Comment:
Why not `var rgba = vec4<f32>(0.0, 0.0, 0.0, uniforms.alpha);` ? The `alpha` in `imageOptions` should only take effect for `numChannels` < 4.